src/routeGeneration/templates/koa.ts

```diff
@@ -88,10 +86,8 @@ export function RegisterRoutes(router: any) {
           const fail = function(error: any) {
               responded++;
               if (responded == security.length && !success) {
-                context.status = error.status || 401;
```

Comment:
Since this failure is due to all of the registered authentication handlers failing, I wonder if it makes sense to keep the default `context.status = error.status || 401`.  Otherwise the authentication middleware will have to make sure to set the status in the error that they throw, since I think Koa will automatically treat those errors as internal.

Reply:
Are you saying that instead of `context.throw(error.status, `, we should have `context.throw(error.status || 401, `? Is there any advantage to setting `context.status` before using `context.throw`?

Reply:
I think so yea, otherwise, the authentication middleware provided will need to know that it must throw an error with a status set.  If one isn't provided, koa will default to 500 which will be misleading from the client perspective.  I think a 401 is a sensible default since, in this particular location, we've tried all of the registered security handlers and none of them succeeded, therefore, returning an unauthorized status seems reasonable.

To your question, from what I can see in the [docs here](https://github.com/koajs/koa/blob/master/docs/api/context.md#ctxthrowstatus-msg-properties) I wouldn't have expected there to be any difference but I also found [this issue](https://github.com/koajs/koa/issues/803) which leads me to think that setting the `context.status` might, unfortunately, be necessary.

Reply:
should be fixed.